src/internal/connector/exchange/service_iterators.go

```diff
@@ -131,31 +130,30 @@ func FilterContainersAndFillCollections(
 			statusUpdater,
 			ctrlOpts,
 		)
-		collections[cID] = &edc
-
-		if edc.State() == data.DeletedState {
-			continue
-		}
 
+		collections[cID] = &edc
 		edc.jobs = append(edc.jobs, jobs...)
 
 		// add the current path for the container ID to be used in the next backup
 		// as the "previous path", for reference in case of a rename or relocation.
 		currPaths[cID] = currPath.String()
 	}
 
-	// any path that wasn't present in the resolver was deleted by the user.
-	// relocations and renames will have removed the dir by id earlier.  What's
-	// left in deletedPaths are only the previous paths that did not appear as
-	// children of the root.
-	for fID, dp := range deletedPaths {
+	// A tombstone is a collection path that needs to be marked for deletion.
+	// Tombstones can occur for a number of reasons: the delta token expired,
+	// the container was deleted in flight, or the user deleted the container
+	// between backup runs.  If events combine to both delete and write content
+	// to the same container (ex: container_1 gets deleted, then container_2
+	// gets created with the same name), it is assumed that the backup consumer
+	// processes deletions before creations, making the combined operation safe.
```

Comment:
>it is assumed that the backup consumer processes deletions before creations, making the combined operation safe.

It's actually a little more nuanced than that. It's that deletions apply to what is essentially the previous path of the container while creations apply to the new path. We're doing state updates now, so the previous path is really a way to find something in the previous state. While relative order of the processing of things makes it more clear, it's not strictly required for correctness as a deletion simply means "don't merge the previous version of this folder with the new version"

This all ties into the [assumptions](https://github.com/alcionai/corso/issues/1884#issuecomment-1361618183) in #1884

Reply:
Granted, but I think the handwave is sufficient for explanation at that placement.